pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -0,0 +1,341 @@
+        self.instr_names = []
+        self.instr_name = None
+        self.instr_group =  defaultdict(list) 
+        self.instr_category = defaultdict(list)
+        self.basic_instr = []
+        self.instr_template = {}
+        
+        self.exclude_reg = []
+        self.include_reg = []
+        
+        self.imm_len = 0
+        self.csr = None
+        self.rs2 = None
+        self.rs1 = None
+        self.rd = None
+        self.imm = BitArray(uint = 0 , length = 32)
+        self.idx = -1
+        self.has_rs1 = 1
+        self.has_rs2 = 1
+        self.has_rd = 1
+        self.has_imm = 1
+
+        self.imm_mask = BitArray(uint = 4294967295 , length = 32)
+        self.is_branch_target = None
+        self.has_label  = 1
+        self.atomic = 0
+        self.branch_assigned = None
+        self.process_load_store = 1
+        self.is_compressed = None
+        self.is_illegal_instr = None
+        self.is_hint_instr = None
+        self.is_floating_point = None
+        self.imm_str = None
+        self.comment = None
+        self.label = None
+        self.is_local_numeric_label = None
+        
+        #Fields Added for debugging These fields are actually from a different files. 
+        self.unsupported_instr = []
+        self.XLEN = 32
+        self.supported_isa  = [riscv_instr_group_t.RV32I]
+        self.implemented_csr = [] 
+    @classmethod
+    def register (cls, instr_name):
+        print("Registering {}".format(instr_name.name))
+        cls.instr_registry[instr_name.name] = 1
+        if instr_name == None:
+            print("\n")
+        return 1
+
+    def create_instr_list(self, cfg):
+        self.instr_names.clear()
+        self.instr_group.clear()
+        self.instr_category.clear()
+        for instr_name, values in self.instr_registry.items():
+
+            if(instr_name in self.unsupported_instr):
+                continue
+            instr_inst = self.create_instr(instr_name) 
+            self.instr_template[instr_name] = instr_inst
+
+            if (not instr_inst.is_supported(cfg)):
+                continue
+            if ((self.XLEN != 32) and (instr_name == "C_JAL")):
+                continue
+            if (("SP" in cfg.reserved_regs) and (instr_name == "C_ADDI16SP")):
+                continue
+            if (cfg.enable_sfence and instr_name == "SFENCE_VMA"):
+                continue
+            if (instr_name in ["FENCE","FENCE_I","SFENCE_VMA"]):
+                continue
+            if (instr_inst.group in self.supported_isa):
+                self.instr_category[instr_inst.category.name].append(instr_name)
+                self.instr_group[instr_inst.group.name].append(instr_name)
+                self.instr_names.append(instr_name)
+                
+        self.build_basic_instruction_list(cfg)
+        self.create_csr_filter(cfg)
+        
+    def create_instr(self,instr_name):
+        instr_inst = eval("rv32i_instr.riscv_"+instr_name+"_instr()")
```

Comment:
This will need to be scaled to be able to create an instance of an instruction from any extended instruction class.
Maybe one option is to use the `instr_name` argument to get the proper group name from the `instr_group` dictionary and then use that return value here. What do you think?

Reply:
@aneels3 could you add a TODO here to ensure that whatever method used here scales to all instruction extensions? We don't have to implement this right now, but it will have to be done in the future, for right now we can get this first PR merged.

Reply:
@udinator Sure!